packages/components/src/pages/VsCodeExtension.vue

```diff
@@ -592,11 +834,160 @@ export default {
     setUserFiltered() {
       this.isUserFiltered = true;
     },
+
+    resetFilters() {
+      this.filters.rootObjects = [];
+      Object.keys(this.filters.declutter).forEach((k) => {
+        this.filters.declutter[k].on = this.filters.declutter[k].default;
+      });
+      this.filters.declutter.hideName.names = [];
+    },
+
+    addHiddenName(name) {
+      const objectName = name.trim();
+
+      if (
+        !objectName ||
+        this.filters.declutter.hideName.names.includes(objectName)
+      ) {
+        return;
+      }
+
+      this.filters.declutter.hideName.names.push(objectName);
+      this.filters.declutter.hideName.on = true;
+      this.clearSelection();
+    },
+
+    removeHiddenName(index) {
+      this.filters.declutter.hideName.names.splice(index, 1);
+      if (this.filters.declutter.hideName.names.length === 0) {
+        this.filters.declutter.hideName.on = false;
+      }
+      this.clearSelection();
+    },
+
+    addRootObject(fqid) {
+      const objectFqid = fqid.trim();
+
+      if (!objectFqid || this.filters.rootObjects.includes(objectFqid)) {
+        return;
+      }
+
+      this.filters.rootObjects.push(objectFqid);
+      this.clearSelection();
+    },
+
+    removeRootObject(index) {
+      this.filters.rootObjects.splice(index, 1);
+      this.clearSelection();
+    },
+
+    filterMediaRequests(events) {
+      const excludedEvents = [];
+      const mediaRegex = [
+        'application/javascript',
+        'application/ecmascript',
+        'audio/.+',
+        'font/.+',
+        'image/.+',
+        'text/javascript',
+        'text/ecmascript',
+        'text/css',
+        'video/.+',
+      ].map((t) => new RegExp(t, 'i'));
+      const mediaFileExtensions = [
+        'aac',
+        'avi',
+        'bmp',
+        'css',
+        'flv',
+        'gif',
+        'htm',
+        'html',
+        'ico',
+        'jpeg',
+        'jpg',
+        'js',
+        'json',
+        'jsonld',
+        'mid',
+        'midi',
+        'mjs',
+        'mov',
+        'mp3',
+        'mp4',
+        'mpeg',
+        'oga',
+        'ogg',
+        'ogv',
+        'ogx',
+        'opus',
+        'otf',
+        'png',
+        'svg',
+        'tif',
+        'tiff',
+        'ts',
+        'ttf',
+        'wav',
+        'weba',
+        'webm',
+        'webp',
+        'woff',
+        'woff2',
+        'xhtml',
+        '3gp',
+        '3g2',
+      ];
+
+      events.forEach((e) => {
+        const eventName = e.codeObject.prettyName;
```

Comment:
It may be more clear to get the request information directly from the event instead of parsing the name.
```ts
const { requestMethod, requestPath } = event;
if (requestMethod === 'GET' && mediaFileExtensions.some((ext) => requestPath.endsWith(ext)) {

}
```

Reply:
We're not checking `mime_type` in this block

Reply:
Finally I get it, fixed https://github.com/applandinc/appmap-js/pull/429/commits/857887c6b2744c42bf9d5402576d28837df4a8f3